### PR TITLE
📊 add issue metrics gh action

### LINKED ISSUE #895: 📊 Add GitHub issue-metrics action
## User Story

As a Data Platform team member
I want to see metrics around how long pull request reviews are taking
So that we can reflect on that data during retros and team check-ins.

## Value

[issue-metrics](https://github.com/github/issue-metrics) will give our team data-driven insights "insights are the keys to success for any software development project. Understanding the health and progress of your issues, pull requests, and discussions is crucial for effective collaboration, maintainership, and project management."
Can see if useful as part check-ins, weekly updates

## Definition of done

- [ ] added to main and support repo
- [ ] raise on our other repositories if useful

## Reference

[How to write good user stories](https://www.gov.uk/service-manual/agile-delivery/writing-user-stories)

### PR DESCRIPTION
**Pull request by bagg3rs**
closes https://github.com/ministryofjustice/data-platform/issues/895

### CODE REVIEW

File: .github/issue-metrics.yml
Comment (jacobwoffenden) [inline .github/issue-metrics.yml:1]: does this need to go into `.github/workflows`?

File: .github/issue-metrics.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+name: Monthly issue metrics
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "3 2 1 * *"
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
name: Monthly issue metrics

on:
  schedule:
    - cron: "3 2 1 * *"
  workflow_dispatch:
```

File: .github/issue-metrics.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+permissions:
+  issues: write
+  pull-requests: read
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
permissions: read-all
```

File: .github/issue-metrics.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  build:
+    name: issue metrics
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
  monthly-issue-metrics:
    name: Monthly issue metrics
    runs-on: ubuntu-latest
    permissions:
      issues: write
      pull-requests: read
    steps:
```

File: .github/issue-metrics.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      - name: Get dates for last month
+        shell: bash
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
      - name: Get dates for last month
        id: get_last_months_dates
        shell: bash
```

File: .github/issue-metrics.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+          # Calculate the last day of the previous month
+          last_day=$(date -d "$first_day +1 month -1 day" +%Y-%m-%d)
+
+          #Set an environment variable with the date range
+          echo "$first_day..$last_day"
+          echo "last_month=$first_day..$last_day" >> "$GITHUB_ENV"
+
+      - name: Run issue-metrics tool
+        uses: github/issue-metrics@v2
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
      - name: Run GitHub Issue Metrics action
        id: run_github_issue_metrics
        uses: github/issue-metrics@66e9f5b342c761af097cc0355053ad1799b43daf # v2.9.0
```

File: .github/issue-metrics.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SEARCH_QUERY: 'repo:ministryofjustice/data-platform is:issue created:${{ env.last_month }} -reason:"not planned"'
+
+      - name: Create issue
+        uses: peter-evans/create-issue-from-file@v4
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]:
```suggestion
      - name: Create issue
        id: create_issue
        uses: peter-evans/create-issue-from-file@433e51abf769039ee20ba1293a088ca19d573b7f # v4.0.1
```

File: .github/issue-metrics.yml [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          title: Monthly issue metrics report
+          token: ${{ secrets.GITHUB_TOKEN }}
+          content-filepath: ./issue_metrics.md
```
Comment (jacobwoffenden) [inline .github/issue-metrics.yml]: Is this a required file too?

**Review by jacobwoffenden — CHANGES_REQUESTED** [review]

File: .github/issue-metrics.yml (hunk 7 above)
Comment (bagg3rs) [reply .github/issue-metrics.yml]: No

File: .github/issue-metrics.yml
Comment (bagg3rs) [reply .github/issue-metrics.yml:1]: 🙃

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**